### DISCUSSION TITLE
使用docker，websocket是如何配置的啊？

### DISCUSSION THREAD

**Discussion by 1052915220** [LagrangeDev/Lagrange.Core#438] [Q&A] [answered]
这是我的配置
![image](https://github.com/LagrangeDev/Lagrange.Core/assets/110186678/f2422061-c12a-4c87-9a63-92bf41e4e63c)
但是用websocket的在线工具连不通。

不使用docker，直接运行就能连通。

能看看你们的配置吗？

**Comment by ghost** [comment]
配置没有问题，你确定这个在线工具能连接你的服务器

**Comment by ghost** [comment]
你可以检查一下
你服务器的防火墙开放对应端口了么？
你 Docker 的对应端口转发了么？
你服务器是外网么？

**Reply by 1052915220** [reply]
我这是自己本地电脑测试的，用的是windows docker desktop。端口8081映射到宿主机了。

**Reply by ghost** [reply]
那应当是无问题的，麻烦提供一下 Trace 日志

---

将 appsettings.json 里的 Logging.LogLevel.Default 设置为 Trace
重新启动

**Reply by 1052915220** [reply]
![image](https://github.com/LagrangeDev/Lagrange.Core/assets/110186678/a63f7849-8e64-46e3-9189-5e6753131489)

**Reply by 1052915220** [reply]
稍等

**Reply by ghost** [reply]
windows 下 Docker 好像需要调整配置才能成功转发端口，你可以搜索一下

**Reply by 1052915220** [reply, marked as answer]
已经解决了，要先登录qq。登录后websocket的这个ForwardWebSocket服务才启动。
![image](https://github.com/LagrangeDev/Lagrange.Core/assets/110186678/686db742-a641-48ff-acc5-bc0bbc4ec248)

**Comment by shenxgan** [comment]
自荐：https://github.com/shenxgan/qqbot

**Reply by shenxgan** [reply]
用的 docker，回复逻辑都是自己代码实现的，容易增加自定义逻辑
